src/ameba/ast/visitors/implicit_return_visitor.cr

```diff
@@ -280,16 +285,24 @@ module Ameba::AST
       false
     end
 
+    def visit(node : Crystal::Block) : Bool
+      @rule.test(@source, node, @stack.positive?, @in_macro)
```

Comment:
don't like an idea of expanding the list of args we pass here, because every rule that uses this visitor will have to be updated. I would suggest to following the existed pattern and wrap the params into a struct or something, similar to what we do for scope visitor and flow expression. Examples:

https://github.com/crystal-ameba/ameba/blob/504017bd7588980ac47992cb3631f885718c73bc/src/ameba/rule/lint/useless_assign.cr#L42

https://github.com/crystal-ameba/ameba/blob/504017bd7588980ac47992cb3631f885718c73bc/src/ameba/rule/lint/unreachable_code.cr#L56